### ISSUE TITLE
Silent crash when using insert to create a large image

### ISSUE THREAD

**Issue by EmilRosen** [jcupitt/pyvips#28]
Hi,
I am trying to stitch a large number of smaller "tiled" images (~7000, 512x512 pixels)  to create one large image (~90000x40000 pixels). The "tiled" images may have overlap and some tiles are missing (these areas should just be black). I tried solving the problem using the code below:
```python
def saveImage(width, height, X, Y, fileNames, outputFileName):
   I = pyvips.Image.black(width, height, bands=3);
   for x, y, fileName in zip(X, Y, fileNames):
      C = pyvips.Image.new_from_file(fileName, access='sequential');

      I = I.insert(C, x, y);

   I.write_to_file(outputFileName);
````

- If I try to copy a small amount of tiled images (~1000) this works perfectly well. However, when I increase the amount of tiled images strange errors occur. 
- When I try to copy all 7000 images, the python program simply stops executing somewhere in the write_to_file function, with no errors given. The written image is corrupt. 
- If I try with an intermediate number of images (~4000), then the saveImage function finishes, but the program stops execution right after that. The written image is sometimes correct, but does sometimes miss large portions of the image. 
- The memory usage is usually around 6 GB before a crash.  
- The same code for creating a 40000x40000 image works perfectly fine. 

If I have understood it correctly, the line `I = I.insert(C, x, y)` creates a new copy of the large image every iteration. I assume this is the culprit somehow, but I have no idea on how to fix it. 

I am using Python 3.6 64 bit version, Windows 10 and have 64 GB RAM.

**Comment by jcupitt** [comment]
Hello Emil,

The problem is that on Windows you can't have more than 2048 files open at once, unfortunately. I think you have two choices:

1. Build your image as a series of passes. You could make your 90k x 40k image and paste 2000 images in, then save the output. Next, close Python, restart it (to make certain all 2000 files have been closed), load the image back, and paste another 2000 in. Once all 7000 are there, save in your output format. I would use vips images (.v) for the intermediates: it's a very fast format which can be loaded on demand.

2. Install linux in a VM and build the image there. I've built 30,000 image mosaics without issue. 

There can also be issues with C stack overflow. libvips uses a recursive algorithm to work out which pixels to calculate next, and for very long pipelines this can overflow and cause a crash. Again, processing in chunks of 2000 at once should fix this.

**Comment by EmilRosen** [comment]
I tried writing in chunks and it works. Thanks! However, I would like to add some notes:

- I did not need to restart python between writing chunks. 
- I had to use a chunk size of 500. If I used a larger chunk size, the first chunk would successfully be written but the program would exit soon after.
- Apparently I had a max of 512 open files. Changing it to 2048 did not allow me to increase the chunk size.
